x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/use_rule_details_flyout.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback } from 'react';
+import { invariant } from '../../../../../common/utils/invariant';
+import type {
+  RuleInstallationInfoForReview,
+  RuleSignatureId,
+} from '../../../../../common/api/detection_engine';
+import type { DiffableRule } from '../../../../../common/api/detection_engine/prebuilt_rules/model/diff/diffable_rule/diffable_rule';
+
+export interface RuleDetailsFlyoutState {
+  flyoutRule: RuleInstallationInfoForReview | null;
+}
+
+export interface RuleDetailsFlyoutActions {
+  openFlyoutForRuleId: (ruleId: RuleSignatureId) => void;
+  closeFlyout: () => void;
+}
+
+export const useRuleDetailsFlyout = (
+  rules: DiffableRule[]
+): RuleDetailsFlyoutState & RuleDetailsFlyoutActions => {
+  const [flyoutRule, setFlyoutRule] = React.useState<RuleInstallationInfoForReview | null>(null);
```

Comment:
This hook, given it knows about `RuleInstallationInfoForReview`, either should not be here in the `rule_management` subdomain, or should be somehow rewritten to remove this dependency. I lean towards just moving it closer to the rule installation table's React context.

Same for the upgrade table's context. But I think there we'd need a slightly modified version of this function (see my next comment about using `ruleId: RuleSignatureId`).